src/Articles/containers/CreateArticle.jsx

```diff
@@ -0,0 +1,321 @@
+      displayProgress: 'none',
+      isFormValid: false,
+      articlePhoto: {
+        name: 'No Photo selected',
+        data: null,
+      },
+    };
+
+    this.handleCreateArticle = this.handleCreateArticle.bind(this);
+
+    this.handleChange = this.handleChange.bind(this);
+
+    this.handleImageUpload = this.handleImageUpload.bind(this);
+  }
+
+  componentWillMount() {
+    const token = localStorage.getItem('user_token');
+    if (token === null) {
+      window.location = '/login';
+    }
+
+    return true;
+  }
+
+  handleCreateArticle = (event) => {
+    event.preventDefault();
+    const {
+      articleTitle, articleBody, articleTags, articlePhoto,
+    } = this.state;
+    const articleData = {
+      article: {
+        title: articleTitle,
+        body: articleBody,
+        description: articleBody,
+      },
+    };
+
+    if (articleTags !== '') {
+      const tags = articleTags.split(',');
+      articleData.tagName = tags;
+    }
+
+    const { postAnArticle } = this.props;
+
+    if (articlePhoto.data !== null) {
+      this.handleImageUpload();
+    } else {
+      postAnArticle(articleData);
+    }
+  }
+
+  handleChange = (event) => {
+    const { target: { name, value, files } } = event;
+    const { isFieldInvalid } = this.state;
+
+    switch (name) {
+      case 'articleTitle':
+        isFieldInvalid.title = !TITLE_REGEX.test(value);
+        break;
+      case 'articleBody':
+        isFieldInvalid.body = !BODY_REGEX.test(value);
+        break;
+      case 'articleTags':
+        isFieldInvalid.tags = !TAGS_REGEX.test(value);
+        break;
+      case 'articlePhoto':
+        this.setState({
+          articlePhoto: {
+            name: 'One Photo Added, click here again to change photo',
+            data: files[0],
+          },
+        });
+        break;
+      default:
+        break;
+    }
+    if (name !== 'articlePhoto') {
+      this.setState({
+        [name]: value,
+      });
+    }
+
+    if ((isFieldInvalid.title === false
+        && isFieldInvalid.body === false
+    )) {
+      this.setState({
+        isFormValid: true,
+      });
+    } else {
+      this.setState({
+        isFormValid: false,
+      });
+    }
+  }
+
+  handleImageUpload = () => {
+    const {
+      articleTitle, articleBody, articleTags, articlePhoto,
+    } = this.state;
+
+    const { postAnArticle } = this.props;
+
+    this.setState({
+      displayProgress: 'block',
+    });
+
+    const imageUpLoad = storage.ref(`article-image/${articlePhoto.data.name}`).put(articlePhoto.data);
+    imageUpLoad.on('state_changed',
+      (snapshot) => {
+        const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+        this.setState({
+          uploadProgress: progress,
+        });
+      },
+      (error) => {
+      },
+      /* istanbul ignore next */
+      (complete) => {
+        storage.ref('article-image').child(articlePhoto.data.name).getDownloadURL()
+          .then(url => {
```

Comment:
Where's the error handling for this?

Reply:
@AGMETEOR kindly note that I have now added the error handler